### PR TITLE
currentworkloadProfileStates - GET

### PR DESCRIPTION
**Pull request by tdaroly**
## ARM API Information (Control Plane)

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [x] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ManagedEnvironments.json
Comment (vinisoto) [inline specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ManagedEnvironments.json:652]: Get all workload Profile States for a Premium Managed Environment

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ManagedEnvironments.json
Comment (vinisoto) [inline specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ManagedEnvironments.json:1015]: workload profile states is not a tracked resource. It is a proxyOnly resource

**Comment by vinisoto** [comment]
Make sure all the checks run successfully

**Review by vinisoto — CHANGES_REQUESTED** [review]

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ManagedEnvironments.json
Comment (tdaroly) [reply specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ManagedEnvironments.json]: image
Changed to `proxyResource`, is that correct ?

**Comment by mentat9** [comment]
@tdaroly - This PR doesn't currently show as needing ARM review, but it's probably because the submission form wasn't filled out. Please do so: you PR cannot be routed and reviewed correctly without that information. If your PR does need ARM review, be sure to study the materials linked in the form and attest to that fact by checking the box. This is required for ARM review.

**Comment by tdaroly** [comment]
> @tdaroly - This PR doesn't currently show as needing ARM review, but it's probably because the submission form wasn't filled out. Please do so: you PR cannot be routed and reviewed correctly without that information. If your PR does need ARM review, be sure to study the materials linked in the form and attest to that fact by checking the box. This is required for ARM review.

**Comment by tdaroly** [comment]
@mentat9 Can you review PR and let me know if anything is missing ? if not, can we merge this ?

**Comment by mentat9** [comment]
> @mentat9 Can you review PR and let me know if anything is missing ? if not, can we merge this ?

Please work with this week's oncall (@rkmanda) to complete ARM review of your PR.

**Comment by vinisoto** [comment]
adding @rkmanda as current ARM review oncall

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ManagedEnvironments.json
Comment (jijohn14) [inline specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ManagedEnvironments.json:1024]: Based on spec, this be `name` of the resource.

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/examples/ManagedEnvironments_ListWorkloadProfileStates.json
Comment (jijohn14) [inline specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/examples/ManagedEnvironments_ListWorkloadProfileStates.json:13]: Update based on spec

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ManagedEnvironments.json
Comment (vinisoto) [reply specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ManagedEnvironments.json:1031]: let's use `instances`

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/examples/ManagedEnvironments_ListWorkloadProfileStates.json
Comment (tdaroly) [reply specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/examples/ManagedEnvironments_ListWorkloadProfileStates.json:17]: done

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ManagedEnvironments.json
Comment (tdaroly) [reply specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ManagedEnvironments.json]: pls re review @jitin

**Comment by mentat9** [comment]
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.App/managedEnvironments/{environmentName}/authtoken": {

POST action name should contain a verb, e.g. getAuthToken

---
Refers to: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ManagedEnvironments.json:647 in 0f2702e. [](commit_id = 0f2702e27a978452928d3c67cde8b8b1356970d0, deletion_comment = False)

**Comment by mentat9** [comment]
>           "404": {

Use default branch for all errors.

---
Refers to: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ManagedEnvironments.json:680 in 0f2702e. [](commit_id = 0f2702e27a978452928d3c67cde8b8b1356970d0, deletion_comment = False)

**Comment by mentat9** [comment]
>             "token": {

Looks like a secret: if so, it needs to be tagged with x-ms-secret extension. Ref: [https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-secret](https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-secret).

---
Refers to: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ManagedEnvironments.json:1050 in 0f2702e. [](commit_id = 0f2702e27a978452928d3c67cde8b8b1356970d0, deletion_comment = False)

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ManagedEnvironments.json
Comment (mentat9) [inline specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ManagedEnvironments.json:1149]: These are duplicates of properties from ProxyResource definition. Also if you have location, this would be a TrackedResource. Recommend removing these and changing the allOf $ref to TrackedResource from v3 common types.

**Comment by mentat9** [comment]
> > @mentat9 Can you review PR and let me know if anything is missing ? if not, can we merge this ?
> 
> Please work with this week's oncall (@rkmanda) to complete ARM review of your PR.

To help load-balance, I reviewed this PR (from ARM perspective). Note however that ARM reviewers don't have permissions to approve or merge PRs. We only review the REST layer for ARM compliance. Azure SDK team owns this repo and the PR assignee is always the person from that team who handles full review, approval and merge.

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ManagedEnvironments.json
Comment (jijohn14) [reply specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ManagedEnvironments.json:1149]: This needs to be proxy resource. Let us remove location for this.

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/examples/ManagedEnvironments_ListWorkloadProfileStates.json
Comment (jijohn14) [inline specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/examples/ManagedEnvironments_ListWorkloadProfileStates.json:13]: fix space

**Comment by mentat9** [comment]
> @mentat9 could you pls review again ? also some of your comment's are not relevant to the changes made by my PR, we have added workitems in our board to fix those. I have used the eyes emoji to reference the one's i am talking about Thanks in advance! :)

ARM signoff is for the whole PR: we can't sign off on just part of it. That said, ARM signoff isn't required to merge changes to an aggregation branch: maybe you should proceed with your PR and wait to get ARM signoff when all the changes flow together to main.

**Comment by ruslany** [comment]
>           "404": {

@mentat9 - can you clarify what you mean by default branch? Are you saying this is redundant because of this ?

 "default": {
            "description": "Common error response.",
            "schema": {
              "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse"
            }
          }

---
In reply to: [1248533237](https://github.com/Azure/azure-rest-api-specs/pull/20550#issuecomment-1248533237) [](http://example.com/codeflow?ancestors=1248533237,1248533237)

---
Refers to: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ManagedEnvironments.json:680 in 0f2702e. [](commit_id = 0f2702e27a978452928d3c67cde8b8b1356970d0, deletion_comment = False)

**Comment by xwang971** [comment]
> > ```
> >         "token": {
> > ```
> 
> Looks like a secret: if so, it needs to be tagged with x-ms-secret extension. Ref: https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-secret.
> 
> Refers to: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ManagedEnvironments.json:1050 in [0f2702e](https://github.com/Azure/azure-rest-api-specs/commit/0f2702e27a978452928d3c67cde8b8b1356970d0). [](commit_id = [0f2702e](https://github.com/Azure/azure-rest-api-specs/commit/0f2702e27a978452928d3c67cde8b8b1356970d0), deletion_comment = False)

Please review this change please https://github.com/Azure/azure-rest-api-specs/pull/20775 

> > ```
> >         "token": {
> > ```
> 
> Looks like a secret: if so, it needs to be tagged with x-ms-secret extension. Ref: https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-secret.
> 
> Refers to: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ManagedEnvironments.json:1050 in [0f2702e](https://github.com/Azure/azure-rest-api-specs/commit/0f2702e27a978452928d3c67cde8b8b1356970d0). [](commit_id = [0f2702e](https://github.com/Azure/azure-rest-api-specs/commit/0f2702e27a978452928d3c67cde8b8b1356970d0), deletion_comment = False)

Hi @mentat9 could I get review on this? https://github.com/Azure/azure-rest-api-specs/pull/20775

**Comment by mentat9** [comment]
> > ```
> >       "404": {
> > ```
> 
> @mentat9 - can you clarify what you mean by default branch? Are you saying this is redundant because of this ?
> 
> "default": { "description": "Common error response.", "schema": { "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse" } }
> 
> In reply to: [1248533237](https://github.com/Azure/azure-rest-api-specs/pull/20550#issuecomment-1248533237)
> 
> Refers to: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ManagedEnvironments.json:680 in [0f2702e](https://github.com/Azure/azure-rest-api-specs/commit/0f2702e27a978452928d3c67cde8b8b1356970d0). [](commit_id = [0f2702e](https://github.com/Azure/azure-rest-api-specs/commit/0f2702e27a978452928d3c67cde8b8b1356970d0), deletion_comment = False)

Yes. Recommend pattern is for error responses to be spec'd on the default response and use the ErrorResponse definition from v3 common types.

**Comment by mentat9** [comment]
> > > ```
> > >         "token": {
> > > ```
> > 
> > 
> > Looks like a secret: if so, it needs to be tagged with x-ms-secret extension. Ref: https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-secret.
> > Refers to: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ManagedEnvironments.json:1050 in [0f2702e](https://github.com/Azure/azure-rest-api-specs/commit/0f2702e27a978452928d3c67cde8b8b1356970d0). [](commit_id = [0f2702e](https://github.com/Azure/azure-rest-api-specs/commit/0f2702e27a978452928d3c67cde8b8b1356970d0), deletion_comment = False)
> 
> Please review this change please #20775
> 
> > > ```
> > >         "token": {
> > > ```
> > 
> > 
> > Looks like a secret: if so, it needs to be tagged with x-ms-secret extension. Ref: https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-secret.
> > Refers to: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ManagedEnvironments.json:1050 in [0f2702e](https://github.com/Azure/azure-rest-api-specs/commit/0f2702e27a978452928d3c67cde8b8b1356970d0). [](commit_id = [0f2702e](https://github.com/Azure/azure-rest-api-specs/commit/0f2702e27a978452928d3c67cde8b8b1356970d0), deletion_comment = False)
> 
> Hi @mentat9 could I get review on this? #20775

Yes, I reviewed and signed off that PR. @tdaroly, can you bring those changes to this PR after they merge?

**Comment by mentat9** [comment]
Please use the instructions in this comment to facilitate ARM review: https://github.com/Azure/azure-rest-api-specs/pull/20550#issuecomment-1248546295.

**Comment by mentat9** [comment]
@tdaroly - This comment is still unaddressed: https://github.com/Azure/azure-rest-api-specs/pull/20550#issuecomment-1251671457.

IIUC, the intent was to apply that feedback in a separate PR, get signoff and merge there and then merge those changes to this PR. If that other PR is ready for review, please provide a link here.

**Comment by mentat9** [comment]
@tdaroly - Approving from ARM. Please in future stop using force push for swagger PRs. Swagger review is already difficult. Wiping out the commit history makes the review job nearly impossible.

**Review by leniatgh — APPROVED** [review]

**Pull request merged**
